src/amber/controller/error.cr

```diff
@@ -7,49 +8,46 @@ module Amber::Controller
       @context.response.content_type = content_type
     end
 
-    def not_found
-      response_format(@ex.message)
+    def bad_request
+      response_format
     end
 
-    def internal_server_error
-      response_format("ERROR: #{internal_server_error_message}")
+    def forbidden
+      response_format
     end
 
-    def forbidden
-      response_format(@ex.message)
+    def not_found
+      response_format
+    end
+
+    def internal_server_error
+      response_format
     end
 
     private def content_type
       if context.request.headers["Accept"]?
         request.headers["Accept"].split(",").first
       else
-        "text/html"
+        "text/plain"
       end
     end
 
-    private def internal_server_error_message
-      # IMPORTANT: #inspect_with_backtrace will fail in some situations which breaks the tests.
-      # Even if you call @ex.callstack you'll notice that backtrace is nil.
-      # #backtrace? is supposed to be safe but it exceptions anyway.
-      # Please don't remove this without verifying that crystal core has been fixed first.
-      @ex.inspect_with_backtrace
-    rescue ex : IndexError
-      @ex.message
-    rescue ex
-      <<-ERROR
-      Original Error: #{@ex.message}
-      Error during 'inspect_with_backtrace': #{ex.message}
-      ERROR
-    end
-
-    private def response_format(message)
+    private def response_format
       case content_type
       when "application/json"
-        {"error": message}.to_json
+        {"error": @ex.message}.to_json
       when "text/html"
-        "<html><body><pre>#{message}</pre></body></html>"
+        html_response
+      else
+        @ex.message
+      end
+    end
+
+    private def html_response
+      if Amber.env.development?
+        Amber::Exceptions::Page.for_runtime_exception(context, @ex).to_s
```

Comment:
This dependency will not exist if the user removes it from the shard.  This needs to go in the template controller so the user can remove it if they want too.